src/stan/math/prim/scal/meta/length_mvt.hpp

```diff
@@ -9,16 +9,12 @@ namespace stan {
 
   // length_mvt() should only be applied to std vector or Eigen matrix
   template <typename T>
-  size_t length_mvt(const T& ) {
-    throw std::out_of_range("length_mvt passed to an unrecognized type.");
```

Comment:
Wouldn't it be preferable to get a compile-time error instead of a run-time error?